versions/development/SPEC.md

```diff
@@ -3091,6 +3092,20 @@ Map[String,Int] xmap = as_map(x) # {"a": 1, "b": 2, "c": 3}
 Map[String,Pair[File,File]] ymap = as_map(y) # {"a": ("a.bam", "a.bai"), "b": ("b.bam", "b.bai")}
 ```
 
+## Array[X] keys(Map[X,Y])
+
+Given an Map ,the `keys` function returns a Array of the keys of map. The order of the keys in the resulting Array is the same as the order of the Pairs in the Map.
+
+In cases where multiple Pairs would produce the same key, the workflow will fail.
```

Comment:
Can you clarify what circumstances would lead to the same key being produced? The nature of a dict would seemingly restrict this from happening